src/pytopmod/core/dcel/operators.py

```diff
@@ -0,0 +1,277 @@
+        )
+
+
+def insert_edge(
+    mesh: dcel_mesh.Mesh,
+    vertex_1_key: VertexKey,
+    edge_1_key: EdgeKey,
+    vertex_2_key: VertexKey,
+    edge_2_key: EdgeKey,
+):
+    """Inserts an edge between two corners.
+
+    If the two corners:
+      - belong to the same face, the face will be split into two new ones.
+      - belong to two different faces, the faces will be merged into a new one.
+    """
+    edge_1_node = mesh.edge_nodes[edge_1_key]
+    edge_2_node = mesh.edge_nodes[edge_2_key]
+
+    # 1.1 - Find the 2nd edges for each corner.
+    edge_1_2_key = (
+        edge_1_node.vertex_1_next_key
+        if edge_1_node.vertex_1_key == vertex_1_key
+        else edge_1_node.vertex_2_next_key
+    )
+    edge_2_2_key = (
+        edge_2_node.vertex_1_next_key
+        if edge_2_node.vertex_1_key == vertex_2_key
+        else edge_2_node.vertex_2_next_key
+    )
+
+    # 1.2 - Find the faces that contain each corner.
+    face_1_key = (
+        edge_1_node.face_1_key
+        if edge_1_node.vertex_1_key == vertex_1_key
+        else edge_1_node.face_2_key
+    )
+    face_2_key = (
+        edge_2_node.face_1_key
+        if edge_2_node.vertex_1_key == vertex_2_key
+        else edge_1_node.face_2_key
+    )
+
+    # Set the vertex and edge information for the new edge.
+    # 2 - Create a new edge node.
+    new_edge_key = mesh.create_edge(
+        vertex_1_key, vertex_2_key, face_1_key, face_2_key, edge_1_2_key, edge_2_2_key
+    )
+    new_edge_node = mesh.edge_nodes[new_edge_key]
+
+    # Non-cofacial insertion.
+    if face_1_key != face_2_key:
+        # 3.1 - Update the face information.
+        # Create a new face.
+        new_face_key = mesh.create_face()
+        # Traverse the corners' faces and replace face_1 and face_2 by the new face.
+        for edge_key in list(
+            itertools.chain(face_trace(mesh, face_1_key), face_trace(mesh, face_2_key))
+        ):
+            edge_node = mesh.edge_nodes[edge_key]
+            if edge_node.face_1_key in (face_1_key, face_2_key):
+                edge_node.face_1_key = new_face_key
+            if edge_node.face_2_key in (face_1_key, face_2_key):
+                edge_node.face_2_key = new_face_key
+        # Set the face information for the new edge.
+        new_edge_node.face_1_key = new_face_key
+        new_edge_node.face_2_key = new_face_key
+        # Delete the old faces.
+        mesh.delete_face(face_1_key)
+        mesh.delete_face(face_2_key)
+
+        # 3.2 - Update the edge information.
+        if edge_1_node.vertex_1_key == vertex_1_key:
+            edge_1_node.vertex_1_next_key = new_edge_key
+        else:
+            edge_1_node.vertex_2_next_key = new_edge_key
+        if edge_2_node.vertex_1_key == vertex_2_key:
+            edge_2_node.vertex_1_next_key = new_edge_key
+        else:
+            edge_2_node.vertex_2_next_key = new_edge_key
+
+    # Cofacial insertion.
+    else:
+        # 4.1 - Update the edge information.
```

Comment:
suggestion: 3.2 and 4.1 steps are identical, we can move them to an helper function

Reply:
Implemented the following utility functions as I think they make the most sense:
- `corner_from_edge_vertex(mesh, edge, vertex)`: Returns the corner defined by an (edge, vertex) pair
- `corner_from_face_vertex(mesh, face, vertex)`: Returns the corner defined by a (face, vertex) pair
- `_replace_face(mesh, old_face, new_face)`: Replaces old_face by new_face in all the edges of old_face
- `_update_next_edge(mesh, edge, at_vertex, new_edge)`: Replaces the next vertex in the passed edge node at the passed vertex.

Note that we also introduce a `Corner` data class that holds a `vertex`, `face` and 2 `edge`s.

I am thinking that in the optic of harmonizing the interfaces between DLFL and DCEL we should implement the same `corner_from_face_vertex` and `corner_from_edge_vertex` functions in DLFL and use that for the operators, I'll work on a PR for that.